### PR TITLE
ARROW-7729: [Python][CI] Pin pandas version to 0.25 in the dask integration test

### PR DESCRIPTION
**Pull request by kszucs**
Additionally test agains dask's latest release not just the master revision.

### CODE REVIEW

**Comment by kszucs** [comment]
@ursabot crossbow submit test-conda-python-3.7-dask-latest test-conda-python-3.8-dask-master

**Comment by kszucs** [comment]
[AMD64 Conda Crossbow Submit (#88781)](https://ci.ursalabs.org/#builders/98/builds/488) builder has been succeeded.

Revision: 083221ce9ac239ffa48f396dd1b94c037da9d634

Submitted crossbow builds: [ursa-labs/crossbow @ ursabot-472](https://github.com/ursa-labs/crossbow/branches/all?query=ursabot-472)

|Task|Status|
|----|------|
|test-conda-python-3.7-dask-latest|[CircleCI](https://circleci.com/gh/ursa-labs/crossbow/tree/ursabot-472-circle-test-conda-python-3.7-dask-latest)|
|test-conda-python-3.8-dask-master|[CircleCI](https://circleci.com/gh/ursa-labs/crossbow/tree/ursabot-472-circle-test-conda-python-3.8-dask-master)|

**Comment by nealrichardson** [comment]
This seems dangerous in the long-term. Is there a Jira for fixing the issue that requires the version pin (at which point we would remove this pin)?

**Comment by kszucs** [comment]
@ursabot crossbow submit test-conda-python-3.7-dask-latest test-conda-python-3.8-dask-master

**Comment by kszucs** [comment]
[AMD64 Conda Crossbow Submit (#88806)](https://ci.ursalabs.org/#builders/98/builds/489) builder has been succeeded.

Revision: b5cb40eec95c1433cc2c6fb33720c5842b6ef94d

Submitted crossbow builds: [ursa-labs/crossbow @ ursabot-473](https://github.com/ursa-labs/crossbow/branches/all?query=ursabot-473)

|Task|Status|
|----|------|
|test-conda-python-3.7-dask-latest|[CircleCI](https://circleci.com/gh/ursa-labs/crossbow/tree/ursabot-473-circle-test-conda-python-3.7-dask-latest)|
|test-conda-python-3.8-dask-master|[CircleCI](https://circleci.com/gh/ursa-labs/crossbow/tree/ursabot-473-circle-test-conda-python-3.8-dask-master)|

**Comment by kszucs** [comment]
@nealrichardson the issue is between dask and pandas, we just pin the right pandas version to test dask arrow compatibility.

**Review by kszucs — COMMENTED** [review]
+1

File: ci/docker/conda-python-dask.dockerfile [hunk 1]
```diff
@@ -23,3 +23,6 @@ FROM ${repo}:${arch}-conda-python-${python}
 ARG dask=latest
 COPY ci/scripts/install_dask.sh /arrow/ci/scripts/
 RUN /arrow/ci/scripts/install_dask.sh ${dask}
+
+# The Spark tests currently break with pandas >= 1.0
```
Comment (jorisvandenbossche) [inline ci/docker/conda-python-dask.dockerfile]: Spark -> dask?

**Review by jorisvandenbossche — COMMENTED** [review]
dask already released a pandas1.0 compatible release a week ago (2.10, https://github.com/dask/dask/releases), so I am surprised this was needed. What was the failure we got?

**Pull request closed without merging**